src/Avalonia.Controls/Platform/ExportAvaloniaModuleAttribute.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System;
+
+namespace Avalonia.Platform
+{
+    [AttributeUsage(AttributeTargets.Assembly, AllowMultiple = true)]
+    public class ExportAvaloniaModuleAttribute : Attribute
+    {
+        public ExportAvaloniaModuleAttribute(string name, Type moduleType)
+        {
+            Name = name;
+            ModuleType = moduleType;
+        }
+
+        public string Name { get; private set; }
+        public Type ModuleType { get; private set; }
+
+        public string RequiredWindowingSubsystem { get; set; } = "";
```

Comment:
What use case do you see for allowing a module to select the windowing/rendering subsystem? I can see this being surprising - "suddenly my app is using GTK on windows and I don't know why!".